File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    lost.sort((a,b) => a -b);
    reserve.sort((a,b) => a -b);
    
    const status = new Array(n).fill(true);
    for (let person of lost) {
        if (reserve.includes(person)) {
            reserve = reserve.filter((val) => val !== person);
            continue;
        }
            
        status[person - 1] = false;
    }
    
    for (let person of lost) {
        if(reserve.includes(person - 1)) {
            reserve = reserve.filter((val) => val !== person - 1);
            status[person - 1] = true;
        } else if (reserve.includes(person + 1)) {
            reserve = reserve.filter((val) => val !== person + 1);
            status[person - 1] = true
        }
    }

    
    const answer = status.reduce((sum, val) => { 
        if(val === true)
            sum += 1;
        return sum }, 0)
    
    return answer;
}

File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    lost.sort()
    reserve.sort()
    
    status = [True]*n
    
    for person in lost :
        if person in reserve:
            reserve.remove(person)
            continue
        
        status[person - 1] = False
    
    for person in lost:
        if person - 1 in reserve:
            reserve.remove(person - 1)
            status[person - 1] = True
        elif person + 1 in reserve:
            reserve.remove(person + 1)
            status[person - 1] = True
            
        
    answer = status.count(True) 

    return answer